### PR TITLE
[VPU][IE TESTS] Support I32 for some eltwise precisions + tests

### PR DESCRIPTION
**Pull request by AndrewBakalinIntel**
Ticket - #-38645
Changes:
 - Add tests for Logical, Comparions eltwises and add new types for eltwise math operations (divide, floor, power, sqdiff)
 - Update asserts in Eltwise stage to support the following combinations: CompareLT - FP16->I32 ,CompareEQ - FP16->I32, LogicalNot - I32->I32, LogicalAnd - I32->I32, Minimum - I32->I32.
 - Update firmware

### CODE REVIEW

**Comment by AndrewBakalinIntel** [comment]
@mikhail-treskin Hi, could you please take a look at the common test changes especially at the first two commits? The details you may find in the desctiption above.
@adudchen @DariaMityagina could you please take a look at the VPU part?

**Review by elatkin — CHANGES_REQUESTED** [review]

File: inference-engine/src/vpu/graph_transformer/src/stages/eltwise.cpp
Comment (DariaMityagina) [inline inference-engine/src/vpu/graph_transformer/src/stages/eltwise.cpp:170]: Is there a specific test for the changed behavior?

File: inference-engine/cmake/vpu_dependencies.cmake
Comment (AndrewBakalinIntel) [reply inference-engine/cmake/vpu_dependencies.cmake]: Done.

File: inference-engine/src/vpu/graph_transformer/src/stages/eltwise.cpp
Comment (AndrewBakalinIntel) [reply inference-engine/src/vpu/graph_transformer/src/stages/eltwise.cpp:170]: Yes, the layer tests with the appropriate precisions are actually targeted for this purpose (otherwise it woud be failed).

**Comment by AndrewBakalinIntel** [comment]
@elatkin could you please also take a look at the changes? It seems like it's required now otherwise the merging button will be unavailable :)

**Review by elatkin — APPROVED** [review]

**Review by mikhail-treskin — CHANGES_REQUESTED** [review]

**Review by DariaMityagina — APPROVED** [review]

**Comment by iefode** [comment]
Similar changes are in #1464

File: inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp
Comment (AndrewBakalinIntel) [reply inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp]: Thanks! Removed.

File: inference-engine/tests/functional/plugin/shared/src/single_layer_tests/comparison.cpp
Comment (AndrewBakalinIntel) [reply inference-engine/tests/functional/plugin/shared/src/single_layer_tests/comparison.cpp]: Nice proposal! Done.

**Comment by AndrewBakalinIntel** [comment]
@mikhail-treskin could you please approve if you don't have any other objections? Otherwise the merge button will be not available :)

**Review by adudchen — APPROVED** [review]

**Review by mikhail-treskin — APPROVED** [review]

**Comment by AndrewBakalinIntel** [comment]
@gladilov-gleb could you please merge?

**Comment by gladilov-gleb** [comment]
@AndrewBakalinIntel 
I don't see from descriptions for which operations you intend to enable I32 support

**Comment by AndrewBakalinIntel** [comment]
> @AndrewBakalinIntel
> I don't see from descriptions for which operations you intend to enable I32 support

I've done this in the ticket, but forgot to mention it there, thanks!

File: inference-engine/src/vpu/graph_transformer/src/stages/eltwise.cpp
Comment (AndrewBakalinIntel) [reply inference-engine/src/vpu/graph_transformer/src/stages/eltwise.cpp:170]: What I was tried to say is that we didn't have such a test before, but now it is tested in a positive way via layer tests,

File: inference-engine/tests/functional/plugin/myriad/shared_tests_instances/single_layer_tests/comparison.cpp
Comment (AndrewBakalinIntel) [reply inference-engine/tests/functional/plugin/myriad/shared_tests_instances/single_layer_tests/comparison.cpp]: Good point, fixed.

File: inference-engine/tests/functional/plugin/myriad/shared_tests_instances/single_layer_tests/comparison.cpp
Comment (AndrewBakalinIntel) [reply inference-engine/tests/functional/plugin/myriad/shared_tests_instances/single_layer_tests/comparison.cpp]: Fixed.

File: inference-engine/tests/functional/plugin/myriad/shared_tests_instances/single_layer_tests/logical.cpp
Comment (AndrewBakalinIntel) [reply inference-engine/tests/functional/plugin/myriad/shared_tests_instances/single_layer_tests/logical.cpp]: Thanks! Done via `skip_tests_config.cpp`.

**Comment by AndrewBakalinIntel** [comment]
@gladilov-gleb @adudchen @elatkin @DariaMityagina we had an offline discussion in our team and we decided not to make the changes into the common part connected with logical eltwises, but create the specific test for Myriad Plugin. Could you please take a look at logical tests once more?
Thanks.

**Review by DariaMityagina — APPROVED** [review]
Took one more look, seems to be fine)

File: inference-engine/tests/functional/plugin/myriad/shared_tests_instances/skip_tests_config.cpp
Comment (elatkin) [inline inference-engine/tests/functional/plugin/myriad/shared_tests_instances/skip_tests_config.cpp:27]: Do you plan to fix this issue about logical-not?

Comment (AndrewBakalinIntel) [reply inference-engine/tests/functional/plugin/myriad/shared_tests_instances/skip_tests_config.cpp:27]: Not me, but it is planned to be fixed for the next release.

Comment (elatkin) [reply inference-engine/tests/functional/plugin/myriad/shared_tests_instances/skip_tests_config.cpp:27]: Do you mean 2021.2 by next release?

Comment (AndrewBakalinIntel) [reply inference-engine/tests/functional/plugin/myriad/shared_tests_instances/skip_tests_config.cpp:27]: Yes.

**Review by elatkin — APPROVED** [review]

**Review by gladilov-gleb — APPROVED** [review]

**Pull request merged**
